### PR TITLE
AVX-36349: QoS class

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: aviatrix/resource_aviatrix_qos_policy.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v3/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAviatrixQosPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceAviatrixQosPolicyCreate,
+		ReadWithoutTimeout:   resourceAviatrixQosPolicyRead,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_qos_policy.go]: This is a list of policies, right? If yes, the resource name is misleading.

File: aviatrix/resource_aviatrix_qos_policy.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+							Description: "QoS class UUID.",
+						},
+						"uuid": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "QoS policy UUID.",
+						},
+					},
+				},
+			},
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_qos_policy.go]: Is "uuid" per policy or per resource?

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_qos_policy.go (hunk 2 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_qos_policy.go]: > Is "uuid" per policy or per resource?

per policy

File: docs/resources/aviatrix_qos_class.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+### Required
+* `name` - (Required) Name of QoS class.
+* `priority` - (Optional) Priority of QoS class.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_qos_class.md]: mention that this accepts an integer.
```suggestion
* `priority` - (Optional) Priority of QoS class. Valid values: (integer)
```

File: docs/resources/aviatrix_qos_class.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+subcategory: "Copilot"
+layout: "aviatrix"
+page_title: "Aviatrix: aviatrix_qos_class"
+description: |-
+  Creates Aviatrix QoS Class
+---
+
+# aviatrix_qos_class
+
+The **aviatrix_qos_class** resource creates the Aviatrix QoS Class.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_qos_class.md]:
```suggestion
The **aviatrix_qos_class** resource creates a class for the Quality of Service (QoS) mechanism in Aviatrix Edge. This feature allows classification based on DSCP value in IP Header.
```

You shouldn't use any abbreviation without first saying what it means.

File: docs/resources/aviatrix_qos_policy_list.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+subcategory: "Copilot"
+layout: "aviatrix"
+page_title: "Aviatrix: aviatrix_qos_policy_list"
+description: |-
+  Creates Aviatrix QoS Policy List
+---
+
+# aviatrix_qos_policy_list
+
+The **aviatrix_qos_policy_list** resource creates the Aviatrix QoS Policy List.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_qos_policy_list.md]:
```suggestion
The **aviatrix_qos_policy_list** resource creates a list of policies (and rules) under the Quality of Service (QoS) mechanism in Aviatrix Edge. This is to be used in conjunction with **aviatrix_qos_class**es to classify based on DSCP value in IP Header.
```

**Review by inc1t3Ful — CHANGES_REQUESTED** [review]

File: docs/resources/aviatrix_qos_policy_list.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  policies {
+    name           = "qos_policy_1"
+    dscp_values    = ["1", "AF11"]
+    qos_class_uuid = "abcd1234"
+  }
+
+  policies {
+    name           = "qos_policy_2"
+    dscp_values    = ["AF22"]
+    qos_class_uuid = "efgh5678"
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_qos_policy_list.md]: i think you can make one of the examples here have a dependency on the QoS class resource `uuid` / `.id` . That way its clear that its not just a random string

Comment (inc1t3Ful) [reply docs/resources/aviatrix_qos_policy_list.md]: be sure to add an example QoS class resource within the snippet as well

**Review by inc1t3Ful — APPROVED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**